### PR TITLE
minor fixes and dependency upgrade

### PR DESCRIPTION
**Pull request by dougEfresh**
* Update dependencies
* add deserialize_str_to_number to convert string to number
* allow native_input and native_output be Optional

### CODE REVIEW

File: src/api/mod.rs [hunk 1]
```diff
@@ -17,8 +17,8 @@ use crate::{common::Cluster, util::rpc_url_from_cluster};
 use solana_client::rpc_client::RpcClient;
 use crate::request_handler::RequestHandler;
 
-const API_URL_V1: &str = "https://api.helius.xyz/v1";
-const API_URL_V0: &str = "https://api.helius.xyz/v0";
+const API_URL_V1: &str = "https://api-mainnet.helius-rpc.com/v1";
+const API_URL_V0: &str = "https://api-mainnet.helius-rpc.com/v0";
```
Comment (dougEfresh) [inline src/api/mod.rs:21]: Taken from typescript sdk 
https://github.com/helius-labs/helius-sdk/blob/main/src/utils/get-helius-endpoints.ts

Comment (bgreni) [reply src/api/mod.rs:21]: Oh interesting I didn't noticed they changed those, thanks!

Could you please update the devnet api url, as well as the rpc endpoints defined in `rpc_url_from_cluster`?

Comment (dougEfresh) [reply src/api/mod.rs:21]: Done

> Oh interesting I didn't noticed they changed those, thanks!
> 
> Could you please update the devnet api url, as well as the rpc endpoints defined in `rpc_url_from_cluster`?

done

Comment (bgreni) [reply src/api/mod.rs:21]: Thank you kindly

**Review by bgreni — APPROVED** [review]

**Comment by bgreni** [comment]
Thank you for the PR! I've gone ahead and merged it in

**Pull request merged**